### PR TITLE
S8 Sorting

### PR DESCRIPTION
**Pull request by kuzivka**

### CODE REVIEW

File: src/store/reducers/listSlice.ts [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
-    farmFilter: number[],
-    rateFilter:number[],
-    priceRange: number[],
+export interface Filters {
+  searchQuery: '';
+  categoryFilter: number | 'all';
+  farmFilter: number[];
+  rateFilter: number[];
+  priceRange: number[];
+  sortBy:
```
Comment (Kost927) [inline src/store/reducers/listSlice.ts]: better to create separate interface or type ISortBy {...}, interface Filters {... sortBy: ISortBy}

File: src/store/reducers/listSlice.ts [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
+    | 'byRate'
+    | 'byNameAZ'
+    | 'nameZA'
+    | 'lowestPrice'
+    | 'highestPrice'
+    | 'unset';
```
Comment (Kost927) [inline src/store/reducers/listSlice.ts]: add to constants or enum

File: src/store/selectors/getFilteredProducts.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import { RootState } from '@store/store';
+import { getPrice } from '@utils/getPrice';
+import { getRatingValue } from '@utils/getRatingValue';
+
+const applySorting = (sortBy: string) => (a: Product, b: Product) => {
+  switch (sortBy) {
+    case 'byRateAsc': {
+      return b.averageRate - a.averageRate;
+    }
+    case 'byRateDsc': {
```
Comment (Kost927) [inline src/store/selectors/getFilteredProducts.ts]: here you can use constants or enum from previous comment

File: src/store/reducers/listSlice.ts (hunk 1 above)
Comment (kuzivka) [reply src/store/reducers/listSlice.ts]: okay

File: src/components/product-list/sort-icon/SortIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+export function SortIcon(props: Object) {
+  return (
+    <SvgIcon sx={sortSelectIcon} viewBox="0 0 7 4" {...props}>
+      <path
+        d="M1.11035 0.900005L3.16535 2.95501C3.25903 3.04813 3.38576 3.1004 3.51785 3.1004C3.64994 3.1004 3.77667 3.04813 3.87035 2.95501L5.87035 0.955005"
+        stroke="#151515"
+        strokeWidth="1.5"
+        strokeLinecap="round"
+        strokeLinejoin="bevel"
+      />
```
Comment (Yuliia143) [inline src/components/product-list/sort-icon/SortIcon.tsx]: Don't like path in JSX

Comment (kuzivka) [reply src/components/product-list/sort-icon/SortIcon.tsx]: removed

**Review by Kost927 — APPROVED** [review]

**Pull request merged**